### PR TITLE
feat(channels): channels package

### PR DESCRIPTION
**Pull request by samuelmasse**
Moves all channels to their own package.

Each channel will have an api that will respect these standards : 

```ts
const telegram = new TelegramChannel()
await telegram.setup(app)

telegram.on('message', async (e) => {
  await telegram.send(e.scope, e.endpoint, { text: 'yoyo' })
})

await telegram.start('gggg', { botToken: 'YOUR_CONFIG' })
```

Closes DEV-2125

### CODE REVIEW

**Review by laurentlp — APPROVED** [review]

File: packages/channels/src/base/channel.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  public has(scope: string) {
+    return this.service.get(scope) !== undefined
+  }
+}
+
+export interface MessageEvent {
+  scope: string
+  endpoint: Endpoint
+  content: any
```
[the remark this answers is not in the document]
Comment (samuelmasse) [reply packages/channels/src/base/channel.ts:80]: Well have better typings for content when the content types get put in a seperate package

File: packages/channels/src/messenger/api.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      const { client } = this.service.get(scope)
+
+      try {
+        const pageId = await client.getPageId()
+        this.pageIdToScope[pageId] = scope
+      } catch {
+        // when in live mode this call can fail. we can work around it for new users since they are supposed to use the botId in the url
+        // we don't show an error because this is correct usage
+      }
+    }
```
Comment (samuelmasse) [inline packages/channels/src/messenger/api.ts:40]: Legacy garbage. Migth stay here forever actually and we just don't bring it over to the next version of the channel

File: packages/channels/src/teams/api.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    const { adapter } = this.service.get(req.scope)
+
+    await adapter.processActivity(req, res, async (turnContext) => {
+      try {
+        const activity = turnContext.activity
+        const convoRef = TurnContext.getConversationReference(activity)
+
+        await this.service.setRef(convoRef.conversation!.id, convoRef)
+        await this.service.receive(
+          req.scope,
```
Comment (samuelmasse) [inline packages/channels/src/teams/api.ts:21]: setRef here doesn't persist to kvs anymore. To be fixed before merging

File: packages/channels/src/teams/service.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  }
+
+  async setRef(key: string, ref: Partial<ConversationReference>) {
+    this.refs[key] = ref
+  }
+
+  async getRef(key: string): Promise<Partial<ConversationReference>> {
+    return this.refs[key]
+  }
```
Comment (samuelmasse) [inline packages/channels/src/teams/service.ts:31]: The be removed and replaced by a Kvs interface

File: packages/channels/src/messenger/client.ts [hunk 5]
```diff
@@ -30,7 +28,7 @@ export class MessengerClient {
         }
       })
     } catch (e) {
-      this.logger.error(e, 'Error occurred trying to setup "getStarted" message')
+      console.error('Error occurred trying to setup "getStarted" message', e)
     }
   }
```
Comment (samuelmasse) [inline packages/channels/src/messenger/client.ts:33]: These console.error were put there temporarly. Replace by a logger interface before merging

File: packages/channels/src/vonage/api.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  private async handleStatusRequest(req: ChannelApiRequest, res: Response) {
+    res.sendStatus(200)
+  }
+
+  private async receive(scope: string, payload: VonageRequestBody) {
+    const identity = payload.to.number
+    const sender = payload.from.number
+
+    const messageContent = payload.message.content
+
```
Comment (samuelmasse) [inline packages/channels/src/vonage/api.ts:36]: Restore support for all content here

**Comment by samuelmasse** [comment]
Index repsonses don't work anymore

**Comment by samuelmasse** [comment]
We should still print webhook paths

File: packages/channels/src/vonage/api.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    let content: ContentType = { type: 'text', text: undefined! }
+    // TODO: Improve Vonage SDK typings
+    switch (messageContent.type as any) {
+      case 'text':
+        const index = Number(messageContent.text)
+        // TODO: handleIndexResponse
+        content = { type: 'text', text: messageContent.text }
```
Comment (samuelmasse) [inline packages/channels/src/vonage/api.ts:44]: Restore index responses

**Review by laurentlp — APPROVED** [review]

File: packages/channels/src/base/channel.ts [hunk 8]
```diff
@@ -1,30 +1,41 @@
...
+export abstract class ChannelTemplate<
   TConfig extends ChannelConfig,
   TService extends ChannelService<TConfig, any>,
   TApi extends ChannelApi<TService>,
   TStream extends ChannelStream<TService>
-> extends Emitter<{
-  message: MessageEvent
-}> {
+> implements Channel
+{
```
[the remark this answers is not in the document]
Comment (samuelmasse) [reply packages/channels/src/base/channel.ts:37]: Yup eslint and prettier disagree what to do on this line

**Pull request merged**